synapse/lib/stormlib/macro.py

```diff
@@ -156,60 +169,71 @@ def getObjLocals(self):
 
-        path = ('cortex', 'storm', 'macros', name)
-        return await self.runt.snap.core.getHiveKey(path)
+        if len(name) > 491:
+            raise s_exc.BadArg(mesg='Macro names may only be up to 491 chars.')
+
+        return self.runt.snap.core.getStormMacro(name, user=self.runt.user)
 
     async def _funcMacroDel(self, name):
-        name = await s_stormtypes.tostr(name)
-        path = ('cortex', 'storm', 'macros', name)
 
-        mdef = await self.runt.snap.core.getHiveKey(path)
-        if mdef is None:
-            mesg = f'Macro name not found: {name}'
-            raise s_exc.NoSuchName(mesg)
+        name = await s_stormtypes.tostr(name)
 
-        user = self.runt.user
-        if mdef['user'] != user.iden and not user.isAdmin():
-            mesg = 'Macro belongs to a different user'
-            raise s_exc.AuthDeny(mesg=mesg)
+        if len(name) > 491:
+            raise s_exc.BadArg(mesg='Macro names may only be up to 491 chars.')
 
-        await self.runt.snap.core.popHiveKey(path)
+        return await self.runt.snap.core.delStormMacro(name, user=self.runt.user)
 
     async def _funcMacroSet(self, name, storm):
         name = await s_stormtypes.tostr(name)
         storm = await s_stormtypes.tostr(storm)
 
-        if len(name) >= 492:
-            raise s_exc.BadArg(mesg='Macro name cannot exceed 491 characters.',
-                               valu=name)
+        if len(name) > 491:
+            raise s_exc.BadArg(mesg='Macro names may only be up to 491 chars.')
 
-        # validation
         await self.runt.getStormQuery(storm)
 
-        path = ('cortex', 'storm', 'macros', name)
+        if self.runt.snap.core.getStormMacro(name) is None:
+            mdef = {'name': name, 'storm': storm}
+            await self.runt.snap.core.addStormMacro(mdef, user=self.runt.user)
+        else:
+            updates = {'storm': storm, 'updated': s_common.now()}
+            await self.runt.snap.core.modStormMacro(name, updates, user=self.runt.user)
+
+    async def _funcMacroMod(self, name, info):
 
-        user = self.runt.user
+        name = await s_stormtypes.tostr(name)
+        info = await s_stormtypes.toprim(info)
 
-        mdef = await self.runt.snap.core.getHiveKey(path)
-        if mdef is not None:
-            if mdef['user'] != user.iden and not user.isAdmin():
-                mesg = 'Macro belongs to a different user'
-                raise s_exc.AuthDeny(mesg=mesg)
+        if len(name) > 491:
+            raise s_exc.BadArg(mesg='Macro names may only be up to 491 chars.')
 
-        mdef = {
-            'user': user.iden,
-            'storm': storm,
-            'edited': s_common.now(),
-        }
+        if not isinstance(info, dict):
+            raise s_exc.BadArg('Macro info must be a dictionary object.')
+
+        for prop in info.keys():
+            if prop not in ('name', 'desc', 'storm'):
+                raise s_exc.BadArg(mesg=f'User may not edit the field: {prop}.')
```

Comment:
1. the storm query may be a heavy storm object and fail to serialize
2. the storm query as a string may be invalid syntax, thus breaking the guarantee we have with `macro set` validating the storm query and resulting in a storm query being stored that won't run